### PR TITLE
OCPBUGS-42807: Preserve metrics when pod crashes

### PR DESCRIPTION
**Pull request by Vincent056**
The metric `file_integrity_operator_node_failed` was being reset after the File Integrity Operator pod restarted, causing the `NodeHasIntegrityFailure` alert to clear despite failures persisting in `fileintegritynodestatuses`. This fix ensures that the operator retrieves the status of existing node failures and re-triggers the alert as needed, maintaining consistency in node failure detection.

### CODE REVIEW

File: tests/e2e/helpers.go [hunk 1]
```diff
@@ -287,6 +287,46 @@ spec:
...
+	}
+	pod := podList.Items[0]
+	t.Logf("Killing operator pod %s", pod.Name)
+	err = f.KubeClient.CoreV1().Pods(namespace).Delete(goctx.TODO(), pod.Name, metav1.DeleteOptions{})
+	if err != nil {
+		return err
+	}
+
+	// wait for the pod to be recreated
+	err = wait.Poll(pollInterval, pollTimeout, func() (bool, error) {
```
Comment (yuumasato) [inline tests/e2e/helpers.go:308]: @Vincent056 We may need to increase the `pollTimeout`.

**Comment by xiaojiey** [comment]
Pre-merge verification pass with https://github.com/openshift/file-integrity-operator/pull/584
steps:
1. Install File Integrity Operator and check "Enable Operator Recommended Monitoring" checkbox.
2. Create FileIntegrity custom resource and wait for the pods to initialize fine.
3. Simulate a failure condition and wait for NodeHasIntegrityFailure alert to stream.
4. Delete the file-integrity-operator pod.
5. After couple of minutes the alert NodeHasIntegrityFailure still exists and the "oc get fileintegritynodestatuses" reports failure for the node on which failure condition was simulated.

**Comment by rhmdnd** [comment]
Looks like we got a failure with the intial delay feature:

```
e2e_test.go:207: Testing that we don't get a ready daemonSet before the initialDelaySeconds
e2e_test.go:210: We should expect a timed out waiting for daemonSet aide-e2e-test-initialdelay 
```

File: pkg/controller/metrics/metrics.go [hunk 2]
```diff
@@ -303,6 +308,33 @@ func (m *Metrics) IncFileIntegrityNodeStatus(condition, node string) {
...
+		return 0, false, fmt.Errorf("failed to get metric for node: %s", node)
+	}
+	c := make(chan prometheus.Metric, 1)
+	ctr.Collect(c)
+	// Ensure the channel is not empty before reading
+	select {
+	case metric := <-c:
+		dto := dto.Metric{}
+		if err := metric.Write(&dto); err != nil {
+			return 0, false, fmt.Errorf("failed to write metric: %w", err)
```
Comment (rhmdnd) [inline pkg/controller/metrics/metrics.go:326]: At first glance, I assumed this method only retrieved metrics based on the name, but it's actually performing a write, which seems more like an update.

Comment (Vincent056) [reply pkg/controller/metrics/metrics.go:326]: no, it is not a write, just need to write out to the channel so we can see the output

**Pull request merged**